GoogleForCreators/web-stories-wp
# Storybook: update knobs to controls

## Context


Storybook has depreciated `knobs` as of 6.4. We want to convert the use of `knobs` to `controls`. We get access to the `controls` add-on via Storybook's essentials addon. ie. `@storybook/addon-essentials`. The Essentials add-on includes the following separate add-on packages:  `controls`, `actions`, `docs`, `viewport`, `background`, `toolbars & globals` and `measure & outline`. 


## Summary


Replaced all uses of Storybook `knobs` to the new `controls` add-on. 

## Relevant Technical Choices


In addition to upgrading Storybook packages, we replaced `@storybook/addon-knobs` with `@storybook/addon-essentials` instead of `@storybook/addon-controls`. This way we can remove the included add-ons. eg.`@storybook/addon-actions`. 

However, we have left the `@storybook/addon-actions` package since there are still several stories that rely on actions. (See todo)

The RTL add-on package is no longer compatible with the upgraded Storybook controls. It is likely that this package is not going to get upgraded, or maintained since it has been [archived in GitHub](https://github.com/ccpu/storybook-rtl-addon). Thus, it has been removed. Created a ticket [#10380](https://github.com/GoogleForCreators/web-stories-wp/issues/10380) to replace this package. 

The new format for creating a Storybook story includes adding any props you want controls on under `args`.  The `argType` key allows you to customize the `arg`. Here is a quick overview of the most common uses. See [Storybook Controls](https://storybook.js.org/docs/react/essentials/controls) for more details. 

```
export default {
  title: 'Directory/To/Story',
  component: MyComponent,
  args: {
    isSomeProp: true,
    uploadError: '',
    someNumberProp: 3,
    listOfOptions: 'some option from a list of options',
  },
  argTypes: {
    onAdd: { action: 'onSubmit fired' },
    onRemove: { action: 'onDelete fired' },
    listOfOptions: {
      // want your control to be a dropdown select? Use options!
      options: someArray,
      control: { type: 'select'} // could list it as 'radio' if you rather.
    },
    someNumberProp: {
      // controls are labeled whatever the props name is, which can be confusing. 
      // You can re-name it here. Now the prop `someNumberProp` will be labeled as `Magic Number`. 
      name: 'Magic Number',
    },
  },
  parameters: {
    controls: {
      // Allows you to filter the props you actually want as controls. You can also use `include`
      exclude: ['someOtherPropWeDontCareAbout'],
    },
  },
};

export const _default = (args) => {
  return <MyComponent {...args} />;
};
```

## To-do



We might want to remove the `actions` add-on. This pr removed all uses of the actions add-on where it was being used in conjunction with the `knobs`.  There are still ~51 files using the `@storybook/addon-actions` package separately. We can remove this package once those file have been updated to use the new `argTypes` parameter.  Ticket #10375 

## User-facing changes


| New Action | New Controls |
| ------------- | ------------- |
|<img width="1023" alt="new action" src="https://user-images.githubusercontent.com/1820266/150858936-b915833f-9fbc-4542-9fd8-a7a17a1346a1.png">|<img width="1023" alt="new controls" src="https://user-images.githubusercontent.com/1820266/150858994-073e017e-9267-4279-b730-2ef405cd9205.png">|




## Testing Instructions






This PR can be tested by following these steps:

1. Run Storybook and see that I didn't break any stories. 
2. The old Knobs are now listed under Controls and should work the same.
3. Actions should also work the same. 


## Reviews

### Does this PR have a security-related impact?
no


### Does this PR change what data or activity we track or use?
no


### Does this PR have a legal-related impact?
no


## Checklist




---



Fixes #8091
